### PR TITLE
feat: TinyMCE improvements [BD-38] [TNL-8779] [BB-4961]

### PR DESCRIPTION
**Pull request by felipetrz**
### Description

Implement TinyMCE UI improvements as specified in [TNL-8779](https://openedx.atlassian.net/browse/TNL-8779)

Depends: https://github.com/edx/edx-platform/pull/29067

### Testing Steps

0. To test this, you'll need:

* A running devstack
* A course, such as the Demonstration Course automatically created by the devstack.

1. Access the discussions application through the link: `http://localhost:2002/discussions/<course_id>`. If using the default demonstration course, the full link for it will be http://localhost:2002/discussions/course-v1:edX+DemoX+Demo_Course.
2. Click the `Add a post` button.

### Things to check

* The toolbar should include the requested buttons.
* Opening the formatting menu or the image and code dialogs should appear with the correct background color, and not transparent.
* The auto-save feature should store a draft every second, which can be restored when the editor is opened with empty content.
* Typing misspelt words should trigger the browser's spellchecker (if enabled in the browser's settings).
* Clicking the Image button on the toolbar should offer accessibility options for decorative images
* The image dialog should also allow uploading an image from the local storage
* The fonts in the editor UI should be correct, and consistent with edX-brand.

### New dependencies

* `raw-loader`: Required to load CSS as raw text, so we can pass it to TinyMCE instead of loading it into the main HTML `<head>`.

### Features In Progress

* Localization support

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @felipetrz! I've created [BLENDED-983](https://openedx.atlassian.net/browse/BLENDED-983) to keep track of it in Jira.   More details are on the [BD-38](https://openedx.atlassian.net/wiki/spaces/COMM/pages/2790490201/BD-38+Discussions+In-Context+Discussion+Moderation+Phase+2) project page.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: src/discussions/posts/data/api.js [hunk 1]
```diff
@@ -146,3 +147,23 @@ export async function deleteThread(threadId) {
...
 }
+
+/**
+ * Upload a file.
+ * @param {Blob} blob The file body
+ * @param {string} filename
+ * @param {string} courseId
+ * @param {string} threadKey
+ * @returns {Promise<{ location: string }>}
+ */
```
Comment (awaisdar001) [inline src/discussions/posts/data/api.js:158]: thanks for the JS docs.

**Review by awaisdar001 — APPROVED** [review]
Thanks, looks good to me.

**Comment by awaisdar001** [comment]
@felipetrz / @xitij2000 Is this good to go?

**Comment by xitij2000** [comment]
> @felipetrz / @xitij2000 Is this good to go?

This is good to go but relies on the upload endpoint. If the upload endpoint changes then this will need to be updated so I didn't merge it yet.

**Comment by awaisdar001** [comment]
> > @felipetrz / @xitij2000 Is this good to go?
> 
> This is good to go but relies on the upload endpoint. If the upload endpoint changes then this will need to be updated so I didn't merge it yet.

Got it, added https://github.com/edx/edx-platform/pull/29067 as a dependency in the description.

**Comment by felipetrz** [comment]
@xitij2000 @awaisdar001 The backend PR has been merged and deployed, this is good to go

**Comment by edx-webhook** [comment]
@felipetrz 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-app-discussions&entry.867055334=https://github.com/edx/frontend-app-discussions/pull/27&entry.1484655318=https://github.com/felipetrz&entry.752974735=2021-10-15+00:05&entry.1917517419=2021-11-04+04:47&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**